src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,95 @@
+package duke.task; 
+
+import java.util.List;
+import duke.DukeException;
+import duke.util.Parser;
+
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public void addTask(String input) throws DukeException {
+        String[] split = input.split(" ", 2);
```

Comment:
Maybe abstract away all the `split` and other string-processing operations into the Parser class. That would make the Tasklist class much simpler and readable.

Reply:
Second this suggestion. Also think that the Parser class should be responsible for checking User input and throwing appropriate Exceptions from there, so that the TaskList can be responsible solely for TaskList related actions without being concerned about the validity of the Task.

Reply:
I see. Thank you for the helpful suggestion. I will update it accordingly 👍